pytensor/tensor/rewriting/subtensor.py

```diff
@@ -469,8 +469,8 @@ def local_subtensor_lift(fgraph, node):
             return [rbcast_subt_x]
 
 
-@register_canonicalize
-@register_specialize
+@register_stabilize("cxx_only")
+@register_canonicalize("cxx_only")
```

Comment:
Let me push against this approach. 

There are two scenarios for this rewrite:
1. It's not very useful, and should be reconsidered regardless of backend
2. It's useful in the context of a larger chain of rewrites

If you start making rewrites exclusive to the C-backend you will forget about 2. But eventually you will want to make numba the default backend and you will want the old tests to pass. You will now have made your task much more challenging because you diverged the C and Numba backends, and the latter's test suite is way more myopic.

It's actually a blessing that Theano/Aesara had very extensive test suites and it was difficult to break things unintentionally. But restricting rewrites to the old well tested backend that we want to eventually replace by the new one, is opting out of this safety net. In a sense you will just be kicking the can down the road. The decision about the rewrite will have to be done regardless, but by then the Numba rewrite passes may look so different that you cannot even reason about the two and make an informed choice.

***

In short I think we should be very very selective about the rewrites that are backend specific. For instance I think we should definitely investigate if the scalarize changes also make sense for the C and JAX backends.